packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -742,12 +774,29 @@ function getHealthCheckCommand(hc: HealthCheck): string[] {
   return hcCommand.concat(cmd);
 }
 
-function renderResourceRequirements(gpuCount: number): CfnTaskDefinition.ResourceRequirementProperty[] | undefined {
-  if (gpuCount === 0) { return undefined; }
-  return [{
-    type: 'GPU',
-    value: gpuCount.toString(),
-  }];
+function renderResourceRequirements(gpuCount: number = 0, inferenceAcceleratorResources: string[] = []):
+CfnTaskDefinition.ResourceRequirementProperty[] | undefined {
+  if (inferenceAcceleratorResources.length > 0 && gpuCount > 0) {
+    throw new Error('Both inference accelerator and gpu count defined in the container definition.');
```

Comment:
```suggestion
    throw new Error('Cannot define both inference accelerator and gpu count in the container definition.');
```
we also need unit test for this.

Reply:
Discussed about this offline but posting here for the record, we decided to render both `gpuCount` and `inferenceAcceleratorResources` properties and let the validators in other stages handle this check.